### PR TITLE
Reinstate missing validations on some of the October frameworks

### PR DESCRIPTION
**Pull request by tekin**
During the October framework onboarding, it was discovered that MISO was not performing some key validations, and that if we were to onboard the frameworks with their full validations, many suppliers would not be able to make their returns. So a decision was made to relax the validation rules on these frameworks for the time being.

We've now had the time to go through these frameworks with Malcolm Eastoe in the MI Team and make decisions on which validations we think should be re-instated. In each case, key validations were turned on, and some secondary validations were also enabled. Each updated definition was run against the existing completed submissions to see how many submissions would have failed using these new validations to judge the likely impact on suppliers ability to make submissions. In cases where there were a large number of failures, or a particular large supplier was consistently failing the rules were relaxed to minimise impact on submissions.

Note this PR only covers just over half the October frameworks: CM/OSG/05/3565, RM1031, RM1070, RM3710, RM3754, RM3767. The rest will be tackled in a separate PR.

(the code used to re-run validations after a definition update can be found [here](https://gist.github.com/tekin/928938657897d516cb7cb4665c0db1bc))

Trello: https://trello.com/c/aSem0DVC/726-re-insert-validations-on-existing-templates

### CODE REVIEW

File: app/models/framework/definition/RM3767.rb [hunk 1]
```diff
@@ -11,7 +11,6 @@ class Invoice < EntryData
 
         field 'Lot Number', :string, exports_to: 'LotNumber', presence: true, lot_in_agreement: true
         field 'Customer URN', :integer, exports_to: 'CustomerURN', urn: true
-        # field 'Tyre Specification', :string, exports_to: 'ProductCode', presence: true
```
Comment (leeky) [inline app/models/framework/definition/RM3767.rb]: As I noted on our phone call, these were fields that existed in the MISO export but were missing from the Excel templates themselves.  I think these can be removed however.

File: app/models/framework/definition/RM1070.rb [hunk 2]
```diff
@@ -14,30 +14,30 @@ class Invoice < EntryData
...
-        field 'Quantity', :string, exports_to: 'UnitQuantity', ingested_numericality: { only_integer: true }, allow_nil: true
+        field 'Invoice Price Per Vehicle', :string, exports_to: 'UnitPrice', ingested_numericality: true
+        field 'Quantity', :string, exports_to: 'UnitQuantity', ingested_numericality: { only_integer: true }
         field 'Total Supplier price including standard factory fit options but excluding conversion costs and work ex VAT', :string, exports_to: 'InvoiceValue', ingested_numericality: true
-        field 'Additional Expenditure to provide goods', :string, exports_to: 'Expenses', ingested_numericality: true, allow_nil: true
-        field 'All Conversion and third party conversion costs excluding factory fit options', :string, exports_to: 'Additional2', ingested_numericality: true, allow_nil: true
+        field 'Additional Expenditure to provide goods', :string, exports_to: 'Expenses', ingested_numericality: true
+        field 'All Conversion and third party conversion costs excluding factory fit options', :string, exports_to: 'Additional2'
```
Comment (CristinaRO) [inline app/models/framework/definition/RM1070.rb]: Was the removal of `ingested_numericality` validation intentional?

(The template still has the shouty "THIS FIELD MUST ONLY CONTAIN NUMBERS TO 2 DECIMAL PLACES").

File: app/models/framework/definition/CM_OSG_05_3565.rb [hunk 3]
```diff
@@ -20,10 +20,10 @@ class Invoice < EntryData
         field 'Item Code', :string, exports_to: 'ProductCode'
         field 'Manufacturers Product Code', :string, exports_to: 'Additional1'
         field 'Unit Quantity', :string, exports_to: 'Additional2'
-        field 'Price per Item', :string, exports_to: 'UnitPrice', ingested_numericality: true, allow_nil: true
+        field 'Price per Item', :string, exports_to: 'UnitPrice', ingested_numericality: true
```
Comment (CristinaRO) [inline app/models/framework/definition/CM_OSG_05_3565.rb]: Can't verify against template, as the file doesn't have any comments about what's mandatory.

Comment (leeky) [reply app/models/framework/definition/CM_OSG_05_3565.rb]: @CristinaRO Generally speaking orange headers are mandatory and blue are optional. CM/OSG/05/3565 is being expired as it's very old and therefore the template isn't as descriptive as the newer ones.

File: app/models/framework/definition/RM1070.rb (hunk 2 above)
Comment (tekin) [reply app/models/framework/definition/RM1070.rb]: oooh, this looks like a mistake. I'll re-run the checks and update.

File: app/models/framework/definition/RM3754.rb [hunk 4]
```diff
@@ -14,14 +21,14 @@ class Invoice < EntryData
...
-        field 'Total Number of Units', :string, exports_to: 'UnitQuantity', ingested_numericality: true, allow_nil: true
+        field 'Product Description', :string, exports_to: 'ProductDescription', presence: true
+        field 'UNSPSC', :string, exports_to: 'UNSPSC', ingested_numericality: { only_integer: true }
+        field 'Unit of Purchase', :string, exports_to: 'UnitType', presence: true
+        field 'Price per Unit', :string, exports_to: 'UnitPrice', ingested_numericality: true
+        field 'Vehicle Registration No', :string, exports_to: 'Additional1', presence: true
+        field 'Total Number of Units', :string, exports_to: 'UnitQuantity', ingested_numericality: true
         field 'Total Charge (ex VAT)', :string, exports_to: 'InvoiceValue', ingested_numericality: true
-        field 'Payment Profile', :string, exports_to: 'Additional2'
+        field 'Payment Profile', :string, exports_to: 'Additional2', case_insensitive_inclusion: { in: PAYMENT_PROFILES }
```
Comment (leeky) [inline app/models/framework/definition/RM3754.rb]: Just an observation - should we consider fixing the case of any entries like this during ingest? (e.g. "monthly" -> "Monthly"). I suspect it'll make the data warehouse easier to query too.

File: app/models/framework/definition/CM_OSG_05_3565.rb (hunk 3 above)
Comment (tekin) [reply app/models/framework/definition/CM_OSG_05_3565.rb]: Confirmed with Malcolm. We should always be validating the unit price as it's one of the core fields.

File: app/models/framework/definition/RM3767.rb [hunk 5]
```diff
@@ -18,11 +18,9 @@ class Invoice < EntryData
         field 'Rim Diameter', :string
         field 'Load Capacity', :string
         field 'Speed Index', :string
-        field 'Customer PostCode', :string, exports_to: 'CustomerPostCode'
```
Comment (leeky) [inline app/models/framework/definition/RM3767.rb]: Good spot! I probably missed these as they are optional fields, so don't throw errors when they're missing!

File: app/models/framework/definition/RM1070.rb (hunk 2 above)
Comment (tekin) [reply app/models/framework/definition/RM1070.rb]: I've put this back to how it used to be.

**Comment by tekin** [comment]
@CristinaRO @leeky I will merge this once the rebased branch passes the build. Will have to leave it to you folks to get this released to production! There is now a separate card for the remaining October frameworks here: https://trello.com/c/PaJYlZ3t/760-reinstate-validations-on-remaining-october-frameworks

**Pull request merged**